input/postgres/explain.go

```diff
@@ -10,9 +10,33 @@ import (
 	"github.com/lib/pq"
 	"github.com/pganalyze/collector/output/pganalyze_collector"
 	"github.com/pganalyze/collector/state"
+	"github.com/pganalyze/collector/util"
 )
 
-func RunExplain(db *sql.DB, connectedDbName string, inputs []state.PostgresQuerySample) (outputs []state.PostgresQuerySample) {
+func RunExplain(server state.Server, inputs []state.PostgresQuerySample, collectionOpts state.CollectionOpts, logger *util.Logger) (outputs []state.PostgresQuerySample) {
+	var samplesByDb = make(map[string]([]state.PostgresQuerySample))
+
+	for _, sample := range inputs {
+		if sample.Database != server.Config.GetDbName() && !contains(server.Config.DbExtraNames, sample.Database) && !server.Config.DbAllNames {
+			continue
+		}
+		samplesByDb[sample.Database] = append(samplesByDb[sample.Database], sample)
+	}
+
+	for dbName, dbSamples := range samplesByDb {
+		db, err := EstablishConnection(server, logger, collectionOpts, dbName)
+
+		if err == nil {
+			dbOutputs := runDbExplain(db, dbSamples)
+			outputs = append(outputs, dbOutputs...)
+
+			db.Close()
```

Comment:
I did this because that's what `stream.go` was doing, but we should be calling `db.Close` regardless of whether or not there is an error, no? Both the `database/sql` and the `lib/pq` docs are vague about this, but if memory serves, `pq` could certainly fail in setup after actually connecting...

Reply:
I think thats a good idea to explicitly close - especially since we enforce a connection limit of 5 connections by the pganalyze collector across the server. Which means if we keep these open and there is multiple DBs, we would quickly exhaust that limit.

Reply:
Oh sure, I agree it should definitely be closed explicitly, I'm just saying I think we should try to close it explicitly regardless of whether or not `EstablishConnection` returns an error (i.e., just because `EstablishConnection` returned an error, that doesn't mean the connection was not at least partly established).

Reply:
Also, looking at this again, in [runner/logs.go](https://github.com/pganalyze/collector/blob/master/runner/logs.go#L37-L44), we log and bail if we can't establish a connection--we should probably log a warning here at least, right?

Reply:
(and the connection argument to `DownloadLogs` is no longer necessary--I'll remove it)

Reply:
>I'm just saying I think we should try to close it explicitly regardless of whether or not EstablishConnection returns an error

I realize now I'm wrong about this because EstablishConnection explicitly returns `nil` for the connection if `err` is set (which may not be the right approach, but that's a separate issue).